File: Codes/3174-clear-digits.py
'''
Author: LetMeFly
Date: 2024-09-05 09:30:56
LastEditors: LetMeFly
LastEditTime: 2024-09-05 09:38:01
'''
class Solution:
    def clearDigits(self, s: str) -> str:
        ans = []
        cntDigit = 0
        for i in range(len(s) - 1, -1, -1):
            if s[i].isdigit():
                cntDigit += 1
            elif cntDigit:
                cntDigit -= 1
            else:
                ans.append(s[i])
        return ''.join(reversed(ans))

File: Codes/3174-clear-digits.java
/*
 * @Author: LetMeFly
 * @Date: 2024-09-05 09:31:08
 * @LastEditors: LetMeFly
 * @LastEditTime: 2024-09-05 13:06:54
 */
class Solution {
    public String clearDigits(String s) {
        StringBuilder ans = new StringBuilder();
        int cntDigit = 0;
        for (int i = s.length() - 1; i >= 0; i--) {
            if (Character.isDigit(s.charAt(i))) {
                cntDigit++;
            }
            else if (cntDigit > 0) {
                cntDigit--;
            }
            else {
                ans.append(s.charAt(i));
            }
        }
        ans.reverse();
        return ans.toString();
    }
}